python/scannerpy/table.py

```diff
@@ -37,13 +37,14 @@ def columns(self, index=None):
         return columns[index] if index is not None else columns
 
     def num_rows(self):
-        return self._descriptor.num_rows
+        return self._descriptor.end_rows[-1]
```

Comment:
These rows are relative to the table itself, not what it samples. So if your table had io items of size 1, end rows would be [1, 2, 3].